catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,700 @@
+        String newFpid;
+        if (config.getFactoryPid()
+                .contains(DISABLED_CONFIGURATION_SUFFIX)) {
+            newFpid = config.getFactoryPid()
+                    .replace(DISABLED_CONFIGURATION_SUFFIX, "");
+        } else {
+            newFpid = config.getFactoryPid()
+                    .concat(DISABLED_CONFIGURATION_SUFFIX);
+        }
+        Dictionary<String, Object> properties = config.getProperties();
+        config.delete();
+        Configuration newConfig = configurationAdmin.createFactoryConfiguration(newFpid, null);
+        newConfig.update(properties);
+        return newConfig;
+    }
+
+    /**
+     * Returns the service binding slots as a map of string properties with the slot name as the key
+     *
+     * @param binding ServiceBindingType to generate the map from
+     * @return A map of service binding slots
+     */
+    private Map<String, Object> getServiceBindingProperties(ServiceBindingType binding) {
+        Map<String, Object> properties = new HashMap<>();
+        for (SlotType1 slot : binding.getSlot()) {
+            List<String> slotValues = slotHelper.getStringValues(slot);
+            if (CollectionUtils.isEmpty(slotValues)) {
+                continue;
+            }
+            properties.put(slot.getName(), slotValues.size() == 1 ? slotValues.get(0) : slotValues);
+        }
+        if (binding.isSetAccessURI() && properties.get(urlBindingName) != null) {
+            properties.put(properties.get(urlBindingName)
+                    .toString(), binding.getAccessURI());
+        }
+        return properties;
+    }
+
+    /**
+     * Gets all the configurations that have a matching registry id
+     *
+     * @param registryId The registry id to match
+     * @return A map of configurations with factory pids as keys
+     * @throws IOException
+     * @throws InvalidSyntaxException
+     */
+    private Map<String, Configuration> getCurrentConfigurations(String registryId)
+            throws IOException, InvalidSyntaxException {
+        Map<String, Configuration> configurationMap = new HashMap<>();
+        Configuration[] configurations = configurationAdmin.listConfigurations(String.format(
+                CONFIGURATION_FILTER,
+                RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+
+        if (configurations == null) {
+            return configurationMap;
+        }
+
+        for (Configuration config : configurations) {
+            configurationMap.put(config.getFactoryPid(), config);
+        }
+
+        return configurationMap;
+    }
+
+    /**
+     * Unmarshal the metacards metadata and extracts the service bindings
+     *
+     * @param metacard Metacard to extract bindings from
+     * @return List of ServiceBindingType
+     * @throws ParserException
+     */
+    private List<ServiceBindingType> getServiceBindings(Metacard metacard) throws ParserException {
```

Comment:
This is very similar to some of what's being done in https://github.com/codice/ddf/pull/817. Maybe we should consider having some kind of common library that does the conversion from `Metacard` to the registry types instead of having all that code spread across so many different modules?